### PR TITLE
stage2: cmp between untyped undefines values results in undef bool

### PR DESCRIPTION
**Pull request by mitchellh**
This makes the `"compare undefined literal with comptime_int"` behavior test pass.

This focused on just fixing this one test, but in the process I found a lot of weird behavior around "undefined" with stage1 (and therefore was unsure what the expected behavior was in stage2). I'll open another issue about that. 

This is my first time making a Zig compiler PR so please let me know if I should run any other commands. The only commands I ran was `test-stage2` and then the stage2 behavior tests runner with `zig2 test ../test/behavior.zig ...`

### CODE REVIEW

File: src/Sema.zig [hunk 1]
```diff
@@ -9244,9 +9244,9 @@ fn cmpSelf(
     const resolved_type = sema.typeOf(casted_lhs);
     const runtime_src: LazySrcLoc = src: {
         if (try sema.resolveMaybeUndefVal(block, lhs_src, casted_lhs)) |lhs_val| {
-            if (lhs_val.isUndef()) return sema.addConstUndef(resolved_type);
+            if (lhs_val.isUndef()) return sema.addConstUndef(Type.initTag(.bool));
```
Comment (andrewrk) [inline src/Sema.zig:9247]:
```suggestion
            if (lhs_val.isUndef()) return sema.addConstUndef(Type.bool);
```

**Review by andrewrk — APPROVED** [review]
Nice! And... derp, silly mistake that I made there. Thanks for the fix!

I'll make this little fixup on merge, no need to push another commit on your end.

File: src/Sema.zig (hunk 1 above)
Comment (mitchellh) [reply src/Sema.zig:9247]: Got it, thanks! You use `initTag(.bool)` one other place in Sema if you want to fix that up too at the same time, that's where I cargo culted this from.

Comment (andrewrk) [reply src/Sema.zig:9247]: Will do :+1:

**Pull request merged**